mempool/clist_mempool.go

```diff
@@ -106,11 +112,23 @@ func (mem *CListMempool) getCElement(txKey types.TxKey) (*clist.CElement, bool)
 	return nil, false
 }
 
-func (mem *CListMempool) getMemTx(txKey types.TxKey) *mempoolTx {
-	if e, ok := mem.getCElement(txKey); ok {
-		return e.Value.(*mempoolTx)
+func (mem *CListMempool) InMempool(txKey types.TxKey) bool {
+	_, ok := mem.getCElement(txKey)
+	return ok
+}
+
+func (mem *CListMempool) addToCache(tx types.Tx) bool {
+	return mem.cache.Push(tx)
+}
+
+func (mem *CListMempool) forceRemoveFromCache(tx types.Tx) {
+	mem.cache.Remove(tx)
+}
+
+func (mem *CListMempool) removeFromCache(tx types.Tx) {
```

Comment:
rename tryRemoveInvalidTxFromCache instead?

Reply:
I renamed it to `tryRemoveFromCache`. The function takes any kind of transaction; it doesn't know that is invalid.